app-modules/ai/src/Models/Scopes/ConfidentialPromptScope.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2025, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Ai\Models\Scopes;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Scope;
+
+class ConfidentialPromptScope implements Scope
+{
+    public function apply(Builder $builder, Model $model): void
+    {
+        if (auth()->user()?->isAdmin) {
+            return;
+        }
+
+        $builder->where('is_confidential', false)->orWhere(function (Builder $query) {
+            $query->where('is_confidential', true)
+                ->where(function (Builder $query) {
+                    $query->where('user_id', auth()->id())
+                        ->orWhereHas('confidentialAccessTeams', function (Builder $query) {
+                            $query->whereHas('users', function (Builder $query) {
+                                $query->where('users.id', auth()->id());
+                            });
+                        })
+                        ->orWhereHas('confidentialAccessUsers', function (Builder $query) {
+                            $query->where('users.id', auth()->id());
+                        });
+                });
+        });
```

Comment:
This seems alright but will need to be updated for `Project`s and should be tested to ensure it is working.

Reply:
Per our conversation this morning, no need to deal with Project right now.

But please add a test for this global scope ensuring that it works.